src/main/java/com/rackspace/jenkins_nodepool/NodePoolSSHLauncher.java

```
@@ -400,7 +404,11 @@ protected void openConnection(final TaskListener tl, final SlaveComputer compute
                     throw ioexception;
                 }
             }
-            Thread.sleep(TimeUnit.SECONDS.toMillis(retryWaitTimeSeconds));
+            try {
+                Thread.sleep(TimeUnit.SECONDS.toMillis(retryWaitTimeSeconds));
+            } catch (InterruptedException ex){
+                Thread.currentThread().interrupt();
```

What does this do? We catch an interrupted exception, but we tell the thread to interrupt again? I've never seen this convention.

---

AFAICT it just sets a flag on the thread that it has been interrupted which can be used higher up the call stack. I don't think it does anything in this case, but is one better than swallowing the exception. 
I Was googling and found various articles like this:
http://www.javapractices.com/topic/TopicAction.do?Id=251

But I'm not entirely sure this is correct, I also haven't use this idiom before.